### PR TITLE
new message validation

### PR DESCRIPTION
**Pull request by mrazafindrazaka**
https://sprints.zoho.eu/team/redspherproduct#itemdetails/P20/I525

### CODE REVIEW

File: structures/message.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const s = require('superstruct')
+const Uuid = s.define('Uuid', require('is-uuid').v4)
+const { ZuluDateTimeStruct } = require('./lib');
+
+const Message = s.type({
+    id: s.defaulted(Uuid, require('uuid').v4()),
+    date: s.defaulted(ZuluDateTimeStruct, new Date().toISOString()),
```
Comment (glickel) [inline structures/message.js:7]:
```suggestion
    date: s.defaulted(ZuluDateTimeStruct, () => new Date().toISOString()),
```

File: structures/message.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    type: s.defaulted(s.enums(['message']), 'message'),
```
Comment (glickel) [inline structures/message.js:8]:
```suggestion
    type: s.defaulted(s.literal('message'), 'message'),
```

File: structures/message.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const s = require('superstruct')
+const Uuid = s.define('Uuid', require('is-uuid').v4)
+const { ZuluDateTimeStruct } = require('./lib');
+
+const Message = s.type({
+    id: s.defaulted(Uuid, require('uuid').v4()),
```
Comment (glickel) [inline structures/message.js:6]:
```suggestion
    id: s.defaulted(Uuid, uuid.v4),
```

Put require at the top `const uuid = require('uuid');

File: structures/message.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const s = require('superstruct')
+const Uuid = s.define('Uuid', require('is-uuid').v4)
```
Comment (glickel) [inline structures/message.js:2]:
```suggestion
const Uuid = s.define('Uuid', isUUid.v4)
```

Do not put require in another location that the top of the code `const isUuid = require('is-uuid');`

File: structures/message.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const s = require('superstruct')
```
Comment (glickel) [inline structures/message.js:1]: unify the code style, finish always by `;`

**Review by glickel — CHANGES_REQUESTED** [review]

**Review by gbo-redspher — CHANGES_REQUESTED** [review]
Can you add some more failing tests ? Like one where you force the id with something that's not an uuid, and another with a wrong date format

Then you're good to go

File: structures/message.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+const s = require('superstruct');
+const { ZuluDateTimeStruct } = require('./lib');
+const uuid = require('uuid');
+const isUuid = require('is-uuid');
+
+const Uuid = s.define('Uuid', isUuid.v4);
```
Comment (gbo-redspher) [inline structures/message.js:6]: I'm pretty sure that you need it as un function when called, but test it first.
```suggestion
const Uuid = s.define('Uuid', isUuid.v4());
```

**Review by gbo-redspher — CHANGES_REQUESTED** [review]
Have a look if at the comment, just tell me if it's ok then I'll approve it

**Review by gbo-redspher — APPROVED** [review]

File: structures/message.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+const Message = s.type({
+    id: s.defaulted(Uuid, uuid.v4()),
```
Comment (glickel) [inline structures/message.js:9]:
```suggestion
    id: s.defaulted(Uuid, uuid.v4),
```
to avoid generating the same uuid every time

**Review by glickel — CHANGES_REQUESTED** [review]

**Review by glickel — APPROVED** [review]

**Pull request merged**